tests/e2e/specs/subscriptions/purchase-sign-up-fee-subscription.spec.js

```diff
@@ -49,6 +56,39 @@ describeif( RUN_SUBSCRIPTIONS_TESTS )(
 			await fillCardDetails( page, card );
 			await shopper.placeOrder();
 			await expect( page ).toMatch( 'Order received' );
+
+			// Get the order ID so we can open it in the merchant view
+			const orderIdField = await page.$(
+				'.woocommerce-order-overview__order.order > strong'
+			);
+			orderId = await orderIdField.evaluate( ( el ) => el.innerText );
+		} );
+
+		it( 'should have a charge for subscription cost with fee', async () => {
+			await merchant.login();
+
+			await merchant.goToOrder( orderId );
+
+			// Pull out and follow the link to avoid working in multiple tabs
+			const paymentDetailsLink = await page.$eval(
+				'p.order_number > a',
+				( anchor ) => anchor.getAttribute( 'href' )
+			);
+			await Promise.all( [
+				page.goto( paymentDetailsLink, {
+					waitUntil: 'networkidle0',
+				} ),
+				uiLoaded(),
+			] );
+
+			await uiLoaded();
```

Comment:
Waiting for `uiLoaded()` once is enough :)

Reply:
Agreed! We've see some flakiness with this in the past, and the fix I found was to wait for the UI to load: https://github.com/Automattic/woocommerce-payments/pull/1543#discussion_r618742823

In this case, I will go ahead and test removing this one here and the one mentioned below, and if we aren't seeing any failures locally in the CI then we can removing the extra `uiLoaded()` check entirely.

Reply:
I removed this and the one below and everything passed :+1: